### DISCUSSION TITLE
How To The Separate Automatic Refetching From Manual One

### DISCUSSION THREAD

**Discussion by vanenshi** [reduxjs/redux-toolkit#2077] [Q&A]
imagine a very common use case: I am using `RefreshControll` in the react-native:

```tsx
const { isFetching, data, refetch } = useQuery();

<RefreshControl
  refreshing={isFetching}
  onRefresh={refetch}
/>;
```
this is totally ok, but the problem is when the cache gets invalid after the 60s, useQuery is going to `refetch` automatically and shows a refresh indicator on the UI, which is wrong.

I am using a very simple solution to solve that and I am looking for a better workaround to manage this state. 
This is my solution:

```tsx
const { isFetching, data, refetch } = useQuery();

const [isFetchingRequestedByUser, setFetchingRequestedByUser] =
  React.useState(false);

useEffect(() => {
  if (!isFetching) {
    setFetchingRequestedByUser(false);
  }
}, [isFetching]);

<RefreshControl
  refreshing={isFetching && isFetchingRequestedByUser}
  onRefresh={() => {
    refetch();
    setFetchingRequestedByUser(true);
  }}
/>;
```

**Comment by phryneas** [comment]
I'm a bit irritated. Why does your cache get invalid after 60s while the item is still being displayed?

**Reply by vanenshi** [reply]
I think I am making a mistake here
Let me ask a question, why after the 60s query automatically start fetching again? (because isFetching gets true)

**Reply by phryneas** [reply]
I can't really answer that. It should only fetch after the data has been removed from the store and then a component requests it again - and it should only do that if the component using the data was not mounted for 60 seconds.

**Reply by vanenshi** [reply]
Do you mean I made a mistake when I am configuring the API? ill check the documentation again and come back here.
thank you for your help

**Reply by vanenshi** [reply]
Ok, I found the problem.
I have a `getCustomers` query with the `customer` cache key, and a `like`, `dislike` query. when every customer gets liked or disliked, I invalidate the `customer` cache. this process makes the `getCustomers` query to refetch the data again.

Now my question is here, how can I separate the automatic refetch from the refetch I manually toggle by refetch functions??

**Reply by phryneas** [reply]
Sorry for the late answer - unfortunately there is not really a way to distinguish the two.
